RecoParticleFlow/PFClusterProducer/interface/PFEcalBarrelRecHitCreator.h

```diff
@@ -0,0 +1,231 @@
+
+  const int maxNrTowers_=2448;
+  const int maxTowerEtaIndex_=17;
+
+ public:  
+  PFEcalBarrelRecHitCreator(const edm::ParameterSet& iConfig,edm::ConsumesCollector& iC):
+    PFRecHitCreatorBase(iConfig,iC)
+    {
+      recHitToken_ = iC.consumes<EcalRecHitCollection>(iConfig.getParameter<edm::InputTag>("src"));
+      srFlagToken_ = iC.consumes<EBSrFlagCollection>(iConfig.getParameter<edm::InputTag>("srFlags"));
+      SREtaSize_ = iConfig.getUntrackedParameter<int> ("SREtaSize",1);
+      SRPhiSize_ = iConfig.getUntrackedParameter<int> ("SRPhiSize",1);
+      crystalsinTT_.resize(maxNrTowers_);
+      TTHighInterest_.resize(maxNrTowers_,0);
+      theTTDetIds_.resize(maxNrTowers_);
+      neighboringTTs_.resize(maxNrTowers_);
+    }
+    
+    void importRecHits(std::unique_ptr<reco::PFRecHitCollection>&out,std::unique_ptr<reco::PFRecHitCollection>& cleaned ,const edm::Event& iEvent,const edm::EventSetup& iSetup) {
+
+      beginEvent(iEvent,iSetup);
+      
+      edm::Handle<EcalRecHitCollection> recHitHandle;
+
+      edm::ESHandle<CaloGeometry> geoHandle;
+      iSetup.get<CaloGeometryRecord>().get(geoHandle);
+  
+      iEvent.getByToken(srFlagToken_,srFlagHandle_);
+
+      // get the ecal geometry
+      const CaloSubdetectorGeometry *gTmp = 
+	geoHandle->getSubdetectorGeometry(DetId::Ecal, EcalBarrel);
+
+      const EcalBarrelGeometry *ecalGeo =dynamic_cast< const EcalBarrelGeometry* > (gTmp);
+
+      iEvent.getByToken(recHitToken_,recHitHandle);
+      for(const auto& erh : *recHitHandle ) {      
+	const DetId& detid = erh.detid();
+	auto energy = erh.energy();
+	auto time = erh.time();
+        bool hi = isHighInterest(detid);
+
+	const CaloCellGeometry * thisCell= ecalGeo->getGeometry(detid);
+  
+	// find rechit geometry
+	if(!thisCell) {
+          throw cms::Exception("ECALCellNotInGeometry")
+	    << "Detid: "<< detid.rawId()<<" not found in geometry";
+	  continue;
+	}
+
+	out->emplace_back(thisCell, detid.rawId(),PFLayer::ECAL_BARREL,
+			   energy); 
+
+        auto & rh = out->back();
+	
+	bool rcleaned = false;
+	bool keep=true;
+
+	//Apply Q tests
+	for( const auto& qtest : qualityTests_ ) {
+	  if (!qtest->test(rh,erh,rcleaned,hi)) {
+	    keep = false;	    
+	  }
+	}
+	  
+	if(keep) {
+	  rh.setTime(time);
+	  rh.setDepth(1);
+	} 
+        else {
+	  if (rcleaned) 
+	    cleaned->push_back(std::move(out->back()));
+          out->pop_back();
+        }
+      }
+    }
+
+    void init(const edm::EventSetup &es) {
+
+      edm::ESHandle<CaloGeometry> pG;
+      es.get<CaloGeometryRecord>().get(pG);   
+
+      edm::ESHandle<EcalTrigTowerConstituentsMap> hetm;
+      es.get<IdealGeometryRecord>().get(hetm);
+      eTTmap_ = &(*hetm);
+  
+      const EcalBarrelGeometry * myEcalBarrelGeometry = dynamic_cast<const EcalBarrelGeometry*>(pG->getSubdetectorGeometry(DetId::Ecal,EcalBarrel));
+      const std::vector<DetId>& vec(myEcalBarrelGeometry->getValidDetIds(DetId::Ecal,EcalBarrel));
```

Comment:
Is this (or anything else in this method) fixed for a given GT/detector configuration or is it changing with IOVs?